ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java

```diff
@@ -148,8 +151,13 @@ public void run() {
                 }
             } finally {
                 end = Instant.now();
-                if (!isRequiredModules) {
-                    evaluator.diagnostics().forEach(this::outputDiagnostic);
+                if (!isMissingModule) {
+                    evaluator.diagnostics().forEach(this::outputDebugDiagnostic);
+                    outputDistinctErrorDiagnostics(evaluator.diagnostics());
+                } else {
+                    if (isContainsError) {
+                        terminal.error("\nCompilation aborted due to errors.");
```

Comment:
Shouldn't we be printing the diagnostics in this case as well?

Reply:
Updated in 4e40545aa33a23cffff5f809782bdf9266d51921